subsys/net/ip/tcp.c

```diff
@@ -33,7 +33,12 @@ LOG_MODULE_REGISTER(net_tcp, CONFIG_NET_TCP_LOG_LEVEL);
 
 static int tcp_rto = CONFIG_NET_TCP_INIT_RETRANSMISSION_TIMEOUT;
 static int tcp_retries = CONFIG_NET_TCP_RETRY_COUNT;
-static int tcp_window = NET_IPV6_MTU;
+static int tcp_window =
+#if (CONFIG_NET_TCP_MAX_RECV_WINDOW_SIZE != 0)
+	CONFIG_NET_TCP_MAX_RECV_WINDOW_SIZE;
+#else
+	(CONFIG_NET_BUF_RX_COUNT * CONFIG_NET_BUF_DATA_SIZE) / 3;
```

Comment:
@rlubos 
Any specific reason behind magic constant 3 here?
Shouldn't it be calculated or at least a KConfig?

Sorry if I'm missing something here...

Reply:
Same reason as for TX window. Note, that if you don't like the default auto calculated value you can always set a specific window size with Kconfig. Or even set the window size at runtime since recently (with socket option).

Reply:
I asked @jukkar regarding TX, but he didn't give any specific reason: 
https://github.com/zephyrproject-rtos/zephyr/issues/23302#issuecomment-1109638891

Yes, that's what we are doing now - explicitly setting window.

That's actually quite interesting to be able to set that with socket, from which version is it supported?
We're on NRF SDK from Nordic, so Zephyr 2.7.99, which doesn't seem to have that

Reply:
It's pretty fresh addition, not part of any official release yet (nor present in NCS):
https://github.com/zephyrproject-rtos/zephyr/pull/44383
https://github.com/zephyrproject-rtos/zephyr/pull/44477